code/modules/clothing/clothing.dm

```diff
@@ -198,6 +200,46 @@ BLIND     // can't see anything
 		var/mob/M = loc
 		M.update_inv_shoes()
 
+/obj/item/clothing/shoes/attackby(obj/item/I, mob/user, params)
+	if(!can_hold_knives)
+		return
+	if(held_knife)
+		user << "<span class='notice'>There's already something in [src].</span>"
+		return
+	if(istype(I, /obj/item/weapon/kitchen/knife) || istype(I, /obj/item/weapon/pen))//can hold both regular pens and energy daggers. made for your every-day tactical librarians/murderers.
+		if(!user.drop_item())
+			return
+		I.loc = src
+		user << "<span class='notice'>You discreetly slip [I] into [src]. Click and drag [src] to yourself to remove it.</span>"
+		held_knife = I
+
+/obj/item/clothing/shoes/MouseDrop(atom/over_object)
+	..()
+	if(!can_hold_knives)
+		return
+	var/mob/M = usr
+	if(M.restrained() || M.stat || !Adjacent(M) || !held_knife)
+		return
+	if(over_object == M)
+		M.put_in_hands(held_knife)
+		usr.visible_message("<span class='warning'>[usr] draws [held_knife] from \his shoes!</span>", "<span class='notice'>You withdraw [held_knife] from [src].</span>")
+		held_knife = null
+	else if(istype(over_object, /obj/screen))
+		switch(over_object.name)
+			if("r_hand")
+				if(!remove_item_from_storage(M))
+					M.unEquip(held_knife)

```

Comment:
as for this i honestly have no clue since i basically copied down paper bin's code for dragging things into your hands. as far as i know it works as intended, and i worry if i remove this something will break, but if you're sure it's not need i'll remove it.